src/aspire/nfft/__init__.py

```diff
@@ -116,13 +138,47 @@ def __new__(cls, *args, **kwargs):
             return super(Plan, cls).__new__(cls)
 
 
-def anufft3(vol_f, fourier_pts, sz, real=False):
-    plan = Plan(sz=sz, fourier_pts=fourier_pts)
+def anufft3(vol_f, fourier_pts, sz, real=False, many=0):
+    """
+    Wrapper for 1, 2, and 3 dimensional Non Uniform FFT Adjoint.
+    Dimension is based on the dimension of fourier_pts and checked against vol_f.
+
+    Selects best availble package from `nfft` `backends` configuration list.
+
+    :paran vol_f: Array representing the signal(s) in Fourier space to be transformed.
+    :param fourier_pts: The points in Fourier space where the Fourier transform is to be calculated,
+            arranged as a dimension-by-K array. These need to be in the range [-pi, pi] in each dimension.
+    :param sz: A tuple indicating the geometry of the signal.
+    :param real: Optional Bool indicating if you would like only the real components, Defaults False.
+    :param many: Optional integer indicating if you would like to compute a batch of `many`
```

Comment:
let's say you have 64 images at 64x64. The data array would be ambiguous with a 64x64x64 volume.

Reply:
I am not sure we can actually detect anything more than we already do unless we enforce Plan+Transform and Plan+Adjoint are single operations s.t. all required information is provided at plan instantiation time.  See the other change I did wrt many~~>ntransforms and the default to 1 logic.

If we went with that approach, besides being a large change to both (cu/fi) implementations we would lose reuse of the Plan+set_pts layer. Currently we can batch batches. This isn't a huge deal, but the reuse is probably one reason why this code was written this way in the first place...

If I'm still missing your point let me know.

Reply:
> let's say you have 64 images at 64x64. The data array would be ambiguous with a 64x64x64 volume.

Right. I forgot that we have to detect the number of dimensions here. If that were fixed (say having a required argument `d` or actually having separate functions `anufft2` and `anufft3``), we would be able to detect the number of transforms, right?

> I am not sure we can actually detect anything more than we already do unless we enforce Plan+Transform and Plan+Adjoint are single operations s.t. all required information is provided at plan instantiation time.

Sorry I'm lost here. You mean skipping the `set_pts` call in `Plan.__init__`?

Reply:
In the anufft and nufft wrapper we can do this. I misunderstood where we were commenting (I thought was in cufinufft.py for some reason). If you want to do it in this PR I can try to have it ready tomorrow morning.

Reply:
I pushed a small change to move this into the anufft/nufft wrappers. The cov2d tutorial exercises that code and seems to repoduce.

Reply:
Ok I see now. We can do it in `anufft3` the input array `sig_f` is 1D for the single case and 2D in the many case. In `nufft3` we can do it because `sz` is specified as an argument. (I didn't realize that.) Having `sz` as an argument is redundant there, since we can infer the shape from the input and the number of dimensions from `fourier_pts`. This will work even in the 64×64×64 case size `fourier_pts` will still be 2×M.

Reply:
Sure, I think we can remove `sz` in the `anufft/nufft` wrappers. I will push that shortly.  I added some color to the other conversation here.

Reply:
`nufft` was as you describe. I pushed it.

I don't believe sz is redundant in `anufft`.  Signal's shape will correspond to fourier_pts there, unless I am missing more details. Thanks

Reply:
> I don't believe sz is redundant in `anufft`.

Yeah you're right. We need it there to specify the size of the output array.